### PR TITLE
#160451461 user can comment on an article

### PR DESCRIPTION
**Pull request by onwuvic**
### What does this PR do?
This PR adds the functionality for a user to make comment on an article

### Description of Task to be completed?
Create a comment input field
A user can save their comment
User comment display immediately if saving was successful

#### How should this be manually tested?
If authenticated, go to a single article and comment

#### Any background context you want to provide?
None
#### What are the relevant pivotal tracker stories?
#160451461

### CODE REVIEW

File: src/components/CommentButton/CommentButton.scss [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    font-size: 15px;
+    font-weight: 600;
+    background-color: #fff;
+    padding: 8px 8px;
+    outline: none;
+    width: 150px;
+    cursor: pointer;
+    margin-top: 20px;
+  }
+}
```
Comment (s-akinrele) [inline src/components/CommentButton/CommentButton.scss]: add new line

File: src/components/CommentDisplayBox/CommentDisplayBox.scss [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+      font-size: 15px;
+      background-color: #fff;
+      padding: 8px 8px;
+      outline: none;
+      width: 100px;
+      cursor: pointer;
+      margin-top: 10px;
+    }
+  }
+}
```
Comment (s-akinrele) [inline src/components/CommentDisplayBox/CommentDisplayBox.scss]: add new line

File: src/containers/ViewAnArticle/ViewArticle.js [hunk 3]
```diff
@@ -20,8 +24,11 @@ class ViewArticle extends Component {
   constructor() {
     super();
     this.state = {
-      success: false, loading: false, failure: false, article: []
```
Comment (onwuvic) [inline src/containers/ViewAnArticle/ViewArticle.js]: This state article is supposed to be an object not an array.

File: src/containers/Profile/Profile.js [hunk 4]
```diff
@@ -141,13 +141,20 @@ class Profile extends Component {
               <Input s={12} l={6} placeholder="First Name" label="Firstname" name="firstName" onChange={this.handleInputChange} value={this.state.firstName} />
               <Input s={12} l={6} placeholder="Last Name" label="Lastname" name="lastName" onChange={this.handleInputChange} value={this.state.lastName}/>
               <Input s={12} l={6} placeholder="Contact" label="Contact" name="contact" onChange={this.handleInputChange} value={this.state.contact}/>
-              <Input s={12} l={6} type='select' label="Gender" name="gender" onChange={this.handleInputChange} value={this.state.gender}>
-                <option disable='disable'>Select </option>
```
Comment (onwuvic) [inline src/containers/Profile/Profile.js]: Replace Input component to normal select tag

File: src/containers/Profile/Profile.js [hunk 5]
```diff
@@ -141,13 +141,20 @@ class Profile extends Component {
...
+                  onChange={this.handleInputChange}
+                  value={this.state.gender}
+                  className="browser-default s6" >
+                  <option value="" disabled>Choose gender</option>
+                  <option value="male">Male</option>
+                  <option value="female">Female</option>
+                </select>
+              </div>
               <Input s={12} type='textarea' rows="4" placeholder="Bio" label="Bio" name="bio" onChange={this.handleInputChange} value={this.state.bio}/>
-              <Button s={12} l={6}className="updateButton" waves='light'>Update Profile  <i className="fas fa-edit"></i></Button>
```
Comment (onwuvic) [inline src/containers/Profile/Profile.js]: Add div around the Button component to style it.

File: src/containers/ViewAnArticle/ViewArticle.js [hunk 6]
```diff
@@ -177,19 +252,25 @@ class ViewArticle extends Component {
...
+  article: PropTypes.object,
+  comments: PropTypes.array,
+  createComment: PropTypes.func,
+  getAllComment: PropTypes.func.isRequired,
+  failure: PropTypes.bool,
   loading: PropTypes.bool,
+  match: PropTypes.object,
   success: PropTypes.bool,
-  failure: PropTypes.bool,
-  articles: PropTypes.object,
```
Comment (onwuvic) [inline src/containers/ViewAnArticle/ViewArticle.js]: Re-arrange the propTypes alphabetically.

File: src/containers/createArticle/CreateArticle.js [hunk 7]
```diff
@@ -268,18 +268,16 @@ class CreateArticle extends Component {
                 handleDrag={this.handleDrag}
                 required
               />
-              <div className="input-field col s12">
+              <div className="col input-field s12">
                 <span htmlFor="Title">Categories</span>
-                <Input
-                  s={12}
-                  type="select"
-                  id="category"
```
Comment (onwuvic) [inline src/containers/createArticle/CreateArticle.js]: Used normal select instead of Input component

File: src/index.js [hunk 8]
```diff
@@ -23,7 +24,7 @@ if (localStorage.token) {
 
 const parsedToken = queryString.parse(window.location.search);
 
-const store = createStore(rootReducer, applyMiddleware(createLogger(), thunk));
```
Comment (onwuvic) [inline src/index.js:26]: Add redux dev tools for easy debugging

File: src/containers/ViewArticle/ViewArticle.js [hunk 9]
```diff
@@ -22,9 +27,12 @@ class ViewArticle extends Component {
   constructor() {
     super();
     this.state = {
-      success: false, loading: false, failure: false, article: []
```
Comment (onwuvic) [inline src/containers/ViewArticle/ViewArticle.js:25]: article is supposed to be an object not array

File: src/containers/ViewArticle/ViewArticle.js [hunk 10]
```diff
@@ -207,23 +283,27 @@ class ViewArticle extends Component {
 }
 
 ViewArticle.propTypes = {
-  viewArticle: PropTypes.func.isRequired,
-  rateArticle: PropTypes.func.isRequired,
-  loading: PropTypes.bool,
```
Comment (onwuvic) [inline src/containers/ViewArticle/ViewArticle.js:212]: Re-arrange prop-types alphabetically.

File: src/containers/ViewArticle/ViewArticle.scss [hunk 11]
```diff
@@ -1,4 +1,5 @@
-@import '../../styles//variables';
```
Comment (onwuvic) [inline src/containers/ViewArticle/ViewArticle.scss:1]: Variable import was not properly imported

**Review by s-akinrele — APPROVED** [review]

**Pull request merged**